hack/generator/pkg/codegen/pipeline/testdata/TestInjectConvertibleInterface-v20211231storage.golden

```diff
@@ -0,0 +1,41 @@
+// Code generated by azure-service-operator-codegen. DO NOT EDIT.
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+package v20211231storage
```

Comment:
Are all 4 of these golden files generated from that 1 test? (I think yes based on the name?). I think we have an item for this already but they should be in their own subfolder.

Reply:
I've worked out how to put them into a subfolder but the change has a wide blast radius due to the change being needed in a shared method from the `test` package. I'll keep the changes out of this PR and will do them separately.